gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala

```diff
@@ -170,7 +172,7 @@ trait HashJoinLikeExecTransformer
   override def outputPartitioning: Partitioning = joinBuildSide match {
     case BuildLeft =>
       joinType match {
-        case _: InnerLike | RightOuter => right.outputPartitioning
+        case _: InnerLike | RightOuter => expandPartitioning(right.outputPartitioning)
```

Comment:
HashJoinLikeExecTransformer is also extended by ShuffledHashJoinExecTransformer. Do we need this fix for SHJ?

Reply:
Yes, I was looking into that, this change will help it there also so I think we can keep it for SHJ also.